mailbox/event/json/src/test/java/org/apache/james/event/json/QuotaEventTest.java

```diff
@@ -21,49 +21,479 @@
-    private static final MailboxListener.QuotaUsageUpdatedEvent QUOTA_USAGE_UPDATED_EVENT =
-        new MailboxListener.QuotaUsageUpdatedEvent(
-            USER,
-            QuotaRoot.quotaRoot("foo", Optional.empty()),
-            Quota.<QuotaCount>builder().used(QuotaCount.count(12)).computedLimit(QuotaCount.count(100)).build(),
-            Quota.<QuotaSize>builder().used(QuotaSize.size(1234)).computedLimit(QuotaSize.size(10000)).build(),
-            Instant.parse("2018-11-13T12:00:55Z"));
-
-    private static final String QUOTA_USAGE_UPDATED_EVENT_JSON =
+    private static final QuotaRoot QUOTA_ROOT = QuotaRoot.quotaRoot("foo", Optional.empty());
+    private static final Quota<QuotaCount> QUOTA_COUNT = Quota.<QuotaCount>builder()
+        .used(QuotaCount.count(12))
+        .computedLimit(QuotaCount.count(100))
+        .build();
+    private static final Quota<QuotaSize> QUOTA_SIZE = Quota.<QuotaSize>builder()
+        .used(QuotaSize.size(1234))
+        .computedLimit(QuotaSize.size(10000))
+        .build();
+    private static final Instant INSTANT = Instant.parse("2018-11-13T12:00:55Z");
+    private static final MailboxListener.QuotaUsageUpdatedEvent DEFAULT_QUOTA_EVENT =
+        new MailboxListener.QuotaUsageUpdatedEvent(USER, QUOTA_ROOT, QUOTA_COUNT, QUOTA_SIZE, INSTANT);
+
+    private static final String DEFAULT_QUOTA_EVENT_JSON =
         "{" +
             "\"QuotaUsageUpdatedEvent\":{" +
+            "\"quotaRoot\":\"foo\"," +
+            "\"countQuota\":{\"used\":12,\"limit\":100,\"limits\":{}}," +
+            "\"time\":\"2018-11-13T12:00:55Z\"," +
+            "\"sizeQuota\":{\"used\":1234,\"limit\":10000,\"limits\":{}}," +
+            "\"user\":\"user\"" +
+            "}" +
+        "}";
+
+    @Nested
+    class WithUser {
+
+        @Nested
+        class WithValidUser {
+
+            @Nested
+            class WithUserContainsOnlyUsername {
+
+                private final MailboxListener.QuotaUsageUpdatedEvent eventWithUserContainsUsername = new MailboxListener.QuotaUsageUpdatedEvent(
+                    User.fromUsername("onlyUsername"),
+                    QUOTA_ROOT,
+                    QUOTA_COUNT,
+                    QUOTA_SIZE,
+                    INSTANT);
+                private final String quotaUsageUpdatedEvent =
+                    "{" +
+                        "\"QuotaUsageUpdatedEvent\":{" +
+                        "\"quotaRoot\":\"foo\"," +
+                        "\"countQuota\":{\"used\":12,\"limit\":100,\"limits\":{}}," +
+                        "\"time\":\"2018-11-13T12:00:55Z\"," +
+                        "\"sizeQuota\":{\"used\":1234,\"limit\":10000,\"limits\":{}}," +
+                        "\"user\":\"onlyUsername\"" +
+                        "}" +
+                    "}";
+
+                @Test
+                void fromJsonShouldReturnQuotaEvent() {
+                    assertThat(QuotaEvent$.MODULE$.fromJson(quotaUsageUpdatedEvent).get())
```

Comment:
it would make sense to create a Java bridge to avoid the use of this ugly syntax everywhere